R/navs-legacy.R

```diff
@@ -0,0 +1,605 @@
+#' @export
+#' @rdname navs
+navs_tab <- function(..., id = NULL, selected = NULL,
+                     header = NULL, footer = NULL) {
+  tabs <- tabsetPanel(
+    ..., type = "tabs", id = id, selected = selected,
+    header = header, footer = footer
+  )
+  fragment(tabs)
+}
+
+#' @export
+#' @rdname navs
+navs_pill <- function(..., id = NULL, selected = NULL,
+                      header = NULL, footer = NULL) {
+  pills <- tabsetPanel(
+    ..., type = "pills", id = id, selected = selected,
+    header = header, footer = footer
+  )
+  fragment(pills)
+}
+
+#' @export
+#' @rdname navs
+navs_pill_list <- function(..., id = NULL, selected = NULL,
+                           header = NULL, footer = NULL,
+                           well = TRUE, fluid = TRUE,
+                           widths = c(4, 8)) {
+  pill_list <- navlistPanel(
+    ..., id = id, selected = selected,
+    header = header, footer = footer,
+    well = well, fluid = fluid,
+    widths = widths
+  )
+  fragment(pill_list)
+}
+
+#' @export
+#' @rdname navs
+navs_hidden <- function(..., id = NULL, selected = NULL,
+                        header = NULL, footer = NULL) {
+  hidden <- tabsetPanel(
+    ..., type = "hidden", id = id, selected = selected,
+    header = header, footer = footer
+  )
+  fragment(hidden)
+}
+
+
+#' @inheritParams shiny::navbarPage
+#' @param bg a CSS color to use for the navbar's background color.
+#' @param inverse Either `TRUE` for a light text color or `FALSE` for a dark
+#'   text color. If `"auto"` (the default), the best contrast to `bg` is chosen.
+#' @export
+#' @rdname navs
+navs_bar <- function(..., title = NULL, id = NULL, selected = NULL,
+                     # TODO: add sticky-top as well?
+                     position = c("static-top", "fixed-top", "fixed-bottom"),
+                     header = NULL, footer = NULL,
+                     bg = NULL, inverse = "auto",
+                     collapsible = TRUE, fluid = TRUE) {
+
+  if (identical(inverse, "auto")) {
+    inverse <- TRUE
+    if (!is.null(bg)) {
+      bg <- htmltools::parseCssColors(bg)
+      bg_contrast <- bs_get_contrast(bs_theme("navbar-bg" = bg), "navbar-bg")
+      inverse <- col2rgb(bg_contrast)[1,] > 127.5
+    }
+  }
+
+  navbar <- navbarPage(
+    title = title, ..., id = id, selected = selected,
+    position = match.arg(position),
+    header = header, footer = footer, collapsible = collapsible,
+    inverse = inverse, fluid = fluid
+  )
+
+  if (!is.null(bg)) {
+    navbar <- tagAppendAttributes(
+      navbar, .cssSelector = ".navbar",
+      style = css(background_color = paste(bg, "!important"))
+    )
+  }
+
+  fragment(navbar, page = bs_page)
+}
+
+
+
+# -----------------------------------------------------------------------
+# 'Internal' tabset logic that was pulled directly from shiny/R/bootstrap.R
+#  (with minor modifications)
+# -----------------------------------------------------------------------
+
+navbarPage <- function(title,
```

Comment:
Rename to `navbar_page`, to make it clear that this is not the same as `shiny::navbarPage`.